### PR TITLE
Implement schema support for disassembly

### PR DESCRIPTION
**Pull request by mcy**
This PR adds a number of new features to the disassembler, foremost the ability to consume a message descriptor that makes the disassembly heuristics much smarter. Changes include:

1.  Given a descriptor, correct disassembly of:
  a. ZigZag values.
  c. Unsigned values.
  b. Bools.
  c. Subnormal floats.
  d. Packed fields.
2. Given a descriptor, remarks with field names and enum value names.
2. Smarter collapse of braces during printing.
3. Better alignment of disassembler remarks.
4. Flags for showing the spec (showing it in -help is rude) and whether to show bitpatterns of floats (off by default now).

### CODE REVIEW

**Comment by mcy** [comment]
(I am probably going to attempt to break this into a sequence of three PRs that improve the printing and flag-parsing facilities of the library, followed by schema-awareness.)

**Comment by mcy** [comment]
This PR is now in a reviewable state.

File: writer.go [hunk 1]
```diff
@@ -358,11 +545,11 @@ func (w *writer) decodeField(src []byte) ([]byte, bool) {
 			}
 			w.Write("\"")
 			delimited = nil
-			goto justBytes
+			goto decodeBytes
 		}
 
 		// Who knows what it is? Bytes or something.
-	justBytes:
+	decodeBytes:
```
Comment (fowles) [inline writer.go:552]: I wonder if tail calls to well named helpers would be easier to follow than gotos here

**Review by fowles — DISMISSED** [review]

**Review by fowles — DISMISSED** [review]

**Review by fowles — APPROVED** [review]

**Comment by Adawntoremember** [comment]
> (I am probably going to attempt to break this into a sequence of three PRs that improve the printing and flag-parsing facilities of the library, followed by schema-awareness.)

**Review by SHASA27 — APPROVED** [review]
Psiphon pro v393

**Review by USDTASTRC555 — APPROVED** [review]

**Pull request merged**